src/main/java/org/opensrp/web/controller/UniqueIdController.java

```diff
@@ -71,37 +71,33 @@
 
 		String message;
 		User user;
-		try {
-			Integer numberToGenerate = Integer.valueOf(getStringFilter("batchSize", request));
-
-			Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-			String currentPrincipalName = authentication.getName();
+		Integer numberToGenerate = Integer.valueOf(getStringFilter("batchSize", request));
+		List<String> idsToPrint = openmrsIdService.getNotUsedIdsAsString(numberToGenerate);
+		SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy-HHmmss");
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		String currentPrincipalName = authentication.getName();
+		String fileName = "QRCodes_".concat(df.format(new Date())).concat("_").concat(currentPrincipalName)
+				.concat("_" + numberToGenerate + ".pdf");
+		response.setHeader("Expires", "0");
+		response.setHeader("Cache-Control", "must-revalidate, post-check=0, pre-check=0");
+		response.setHeader("Pragma", "public");
+		response.setContentType("application/pdf");
+		response.setHeader("Content-Disposition", "attachment; filename=" + fileName);
+
+		try (ByteArrayOutputStream byteArrayOutputStream = PdfUtil.generatePdf(idsToPrint, 140, 140, 1, 5);
+		     FileOutputStream fileOutputStream = new FileOutputStream(qrCodesDir + File.separator + fileName);
+		     OutputStream os = response.getOutputStream();) {
 			user = openmrsUserService.getUser(currentPrincipalName);
 			if (!checkRoleIfRoleExitst(user.getRoles(), "opensrp-generate-qr-code")) {
 				return new ResponseEntity<>("Sorry, insufficient privileges to generate ID QR codes", HttpStatus.OK);
 			}
 
 			openmrsIdService.downloadAndSaveIds(numberToGenerate, currentPrincipalName);
-			List<String> idsToPrint = openmrsIdService.getNotUsedIdsAsString(numberToGenerate);
-			SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy-HHmmss");
-
-			String fileName = "QRCodes_".concat(df.format(new Date())).concat("_").concat(currentPrincipalName)
-					.concat("_" + numberToGenerate + ".pdf");
-			ByteArrayOutputStream byteArrayOutputStream = PdfUtil.generatePdf(idsToPrint, 140, 140, 1, 5);
 			if (byteArrayOutputStream.size() > 0) {
 				//mark ids as used
-				FileOutputStream fileOutputStream = new FileOutputStream(qrCodesDir + File.separator + fileName);
 				fileOutputStream.write(byteArrayOutputStream.toByteArray());
 				fileOutputStream.close();
 				openmrsIdService.markIdsAsUsed(idsToPrint);
-
-				response.setHeader("Expires", "0");
-				response.setHeader("Cache-Control", "must-revalidate, post-check=0, pre-check=0");
-				response.setHeader("Pragma", "public");
-				response.setContentType("application/pdf");
-				response.setHeader("Content-Disposition", "attachment; filename=" + fileName);
-
-				OutputStream os = response.getOutputStream();
 				byteArrayOutputStream.writeTo(os);
 				os.flush();
 				os.close();
```

Comment:
Remove this as it should be closed automatically

Reply:
Do I need to remove os.flush() as well, as the stream will be flushed automatically on close. Please confirm @githengi .